src/dev.ts

```diff
@@ -0,0 +1,279 @@
+      const contractFiles = (entries as unknown as string[]).filter((f) =>
+        typeof f === 'string' && f.endsWith('.js')
+      )
+      this.metrics.contractsWatched = contractFiles.length
+
+      console.log(colors.green(`✅ Watching ${contractFiles.length} contract files for changes`))
+    } catch (error) {
+      console.error(colors.red('❌ Error setting up file watching:'), error)
+    }
+  }
+
+  private handleContractChange (eventType: string, filename: string) {
+    const changeEvent: ContractChangeEvent = {
+      file: filename,
+      type: eventType === 'rename' ? 'added' : 'changed',
+      timestamp: new Date()
+    }
+
+    this.contractChanges.push(changeEvent)
+    this.metrics.changesDetected++
+
+    console.log(colors.yellow(`📝 Contract ${eventType}: ${filename}`))
+
+    if (this.options['hot-reload'] !== false) {
+      this.triggerHotReload(filename)
+    }
+  }
+
+  private triggerHotReload (filename: string) {
+    try {
+      console.log(colors.blue(`🔄 Hot reloading contract: ${filename}`))
+
+      // For hot reload, we need to:
+      // 1. Re-pin the changed contract (if it's a source file)
+      // 2. Redeploy to the running server
+      // 3. Notify the dashboard of the change
+
+      if (filename.endsWith('.js') && !filename.includes('manifest')) {
+        console.log(colors.gray('   → Triggering contract redeployment...'))
+        // The server's contract preloading will handle the actual redeployment
+        // We just need to signal that contracts have changed
+        this.metrics.hotReloads++
+        console.log(colors.green('   ✅ Hot reload triggered'))
+      }
+    } catch (error) {
+      console.error(colors.red('❌ Hot reload failed:'), error)
+    }
+  }
+
+  private async startInteractiveMode () {
+    console.log(colors.blue('🎛️  Starting interactive mode...'))
+    console.log(colors.gray('Interactive commands:'))
+    console.log(colors.gray('  - Press Ctrl+C to exit'))
+    console.log(colors.gray('  - File changes will trigger hot reload automatically'))
+    console.log(colors.gray('  - Check the dashboard for live contract status'))
+    console.log()
+
+    // Setup graceful shutdown
+    process.on('SIGINT', () => {
+      console.log(colors.yellow('\n🛑 Shutting down development environment...'))
+      this.cleanup()
+      process.exit(0)
+    })
+
+    // Keep the process alive
+    console.log(colors.green('🎯 Live development environment ready!'))
+    console.log(colors.gray('Watching for contract changes...'))
+
+    // Keep the process alive indefinitely
+    await new Promise(() => {})
+  }
+
+  private cleanup () {
+    console.log(colors.blue('🧹 Cleaning up watchers...'))
+    for (const watcher of this.watchers) {
+      if (watcher && typeof watcher.close === 'function') {
+        watcher.close()
+      }
+    }
+    this.watchers = []
+  }
+
+  private printStatus () {
+    const uptime = Math.floor((Date.now() - this.startTime.getTime()) / 1000)
```

Comment:
`Date` will not give you accurate uptime calculations. This ought to be done with `performance.now`.